### PR TITLE
TLF-99 Updating Legal representative's details

### PR DESCRIPTION
**Pull request by dk4g**
## What? 
TLF-99 - Added Updating Legal representative's details page [TLF-99](https://collaboration.homeoffice.gov.uk/jira/browse/TLF-99)
## Why? 
So that UKVI have current and accurate records of the legal representative
## How? 
Added the relevant fields, validation rules on the legal representative details page and displayed provided details on Check your answers page.
## Check list

- [x] I have reviewed my own pull request
- [ ] I have written tests (if relevant)

### CODE REVIEW

**Review by vinodhasamiyappanHO — APPROVED** [review]
Looks good to me

File: apps/coa/translations/src/en/validation.json
Comment (RobertMcCann) [inline apps/coa/translations/src/en/validation.json:1]: Do we have any information on what the validation messages should be for `checkSRANum`  and `checkOISCNum` ? 

These messages have a couple mistakes in them and i couldn't find what the content should say?

File: apps/coa/fields/index.js [hunk 1]
```diff
@@ -327,6 +327,35 @@ module.exports = {
     formatter: ['ukPostcode'],
     className: ['govuk-input', 'govuk-input--width-10']
   },
+  'legal-company-name': {
+    validate: ['notUrl'],
```
Comment (RobertMcCann) [inline apps/coa/fields/index.js:331]: Maybe legal-company-name should have a maxlength validator?

**Review by MikeMaratheftis — APPROVED** [review]
Code looks solid! Added validation for OISC or SRA numbers, tweaked error messages, and updated translations. Couldn't find any areas to improve or fix. Great job!

File: apps/coa/translations/src/en/validation.json
Comment (vinodhasamiyappanHO) [reply apps/coa/translations/src/en/validation.json:1]: Changes done with the updated error message.

File: apps/coa/fields/index.js (hunk 1 above)
Comment (vinodhasamiyappanHO) [reply apps/coa/fields/index.js]: Added maxlength validation for company name to allow upto 200 characters

**Review by dk4g — COMMENTED** [review]
looks good to me

**Pull request merged**